### PR TITLE
You tube search and unit test assignment

### PR DESCRIPTION
**Pull request by suyash929**
@sriram15690 Please review YouTube Search assignment and Unit test implementation as well

### CODE REVIEW

File: coverage/Chrome 88.0.4324.146 (Windows 10)/base.css [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+.missing-if-branch .typ, .skip-if-branch .typ {
+    color: inherit !important;
+}
+.coverage-summary {
+  border-collapse: collapse;
+  width: 100%;
+}
+.coverage-summary tr { border-bottom: 1px solid #bbb; }
+.keyline-all { border: 1px solid #ddd; }
+.coverage-summary td, .coverage-summary th { padding: 10px; }
```
Comment (sriram15690) [inline coverage/Chrome 88.0.4324.146 (Windows 10)/base.css:122]: Did u try .sass for writing CSS?

File: coverage/Chrome 88.0.4324.146 (Windows 10)/block-navigation.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/* eslint-disable */
+var jumpToCode = (function init() {
+    // Classes of code we would like to highlight in the file view
+    var missingCoverageClasses = ['.cbranch-no', '.cstat-no', '.fstat-no'];
```
Comment (sriram15690) [inline coverage/Chrome 88.0.4324.146 (Windows 10)/block-navigation.js:4]: try using const, let

File: coverage/Chrome 88.0.4324.146 (Windows 10)/block-navigation.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    function goToNext() {
+        var nextIndex = 0;
+
+        if (
+            typeof currentIndex === 'number' &&
+            currentIndex < missingCoverageElements.length - 1
+        ) {
+            nextIndex = currentIndex + 1;
```
Comment (sriram15690) [inline coverage/Chrome 88.0.4324.146 (Windows 10)/block-navigation.js:59]: u can use += 1

File: coverage/Chrome 88.0.4324.146 (Windows 10)/helper.js.html [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+<span class="cline-any cline-yes">3x</span>
+<span class="cline-any cline-yes">1x</span>
+<span class="cline-any cline-yes">2x</span>
+<span class="cline-any cline-yes">1x</span>
+<span class="cline-any cline-yes">1x</span>
+<span class="cline-any cline-yes">1x</span>
+<span class="cline-any cline-neutral">&nbsp;</span>
+<span class="cline-any cline-neutral">&nbsp;</span>
+<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">function timeDifference(current, previous) {
+&nbsp;
```
Comment (sriram15690) [inline coverage/Chrome 88.0.4324.146 (Windows 10)/helper.js.html:147]: JS inside SPAN?

File: coverage/Chrome 88.0.4324.146 (Windows 10)/base.css (hunk 1 above)
Comment (suyash929) [reply coverage/Chrome 88.0.4324.146 (Windows 10)/base.css:122]: No I haven't used it, as we were asked to use CSS only. The above file is automatically generated when we run karma for unit tests. It generates a html file where we can se details about coverage.

File: searchYoutubeVideo/pagination.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  load();
+  let start = items*currentPage;
+  let end = start + items;
+  let paginatedItems = data.slice(start,end);
+
+  for(let i = 0;i < paginatedItems.length ;i++){
+    let duration = timeDifference(new Date(),new Date(paginatedItems[i].snippet.publishedAt));
+    let view = numFormatter(paginatedItems[i].statistics.viewCount);
+
+    let thumbnailContent = `<div class='thumbnail-box'>
```
Comment (sriram15690) [inline searchYoutubeVideo/pagination.js:30]: html creation in JS? Are we not using any frameworks liek React?

**Pull request merged**